test/test_functional_tensor.py

```diff
@@ -692,6 +433,250 @@ def test_scale_channel(self):
         assert_equal(scaled_cpu, scaled_cuda.to('cpu'))
 
 
+class TestAffine:
+
+    @pytest.mark.parametrize('device', cpu_and_gpu())
+    @pytest.mark.parametrize('height, width', [(26, 26), (32, 26)])
+    @pytest.mark.parametrize('dt', [None, torch.float32, torch.float64, torch.float16])
+    def test_identity_map(self, device, height, width, dt):
+        # Tests on square and rectangular images
+        scripted_affine = torch.jit.script(F.affine)
+        tensor, pil_img = _create_data(height, width, device=device)
+
+        if dt == torch.float16 and device == "cpu":
+            # skip float16 on CPU case
+            return
+
+        if dt is not None:
+            tensor = tensor.to(dtype=dt)
+
+        # 1) identity map
+        out_tensor = F.affine(tensor, angle=0, translate=[0, 0], scale=1.0, shear=[0.0, 0.0], interpolation=NEAREST)
+
+        assert_equal(tensor, out_tensor, msg="{} vs {}".format(out_tensor[0, :5, :5], tensor[0, :5, :5]))
+        out_tensor = scripted_affine(
+            tensor, angle=0, translate=[0, 0], scale=1.0, shear=[0.0, 0.0], interpolation=NEAREST
+        )
+        assert_equal(tensor, out_tensor, msg="{} vs {}".format(out_tensor[0, :5, :5], tensor[0, :5, :5]))
+
+    @pytest.mark.parametrize('device', cpu_and_gpu())
+    @pytest.mark.parametrize('height, width', [(26, 26)])
+    @pytest.mark.parametrize('dt', [None, torch.float32, torch.float64, torch.float16])
+    @pytest.mark.parametrize('angle, config', [
+        (90, {'k': 1, 'dims': (-1, -2)}),
+        (45, None),
+        (30, None),
+        (-30, None),
+        (-45, None),
+        (-90, {'k': -1, 'dims': (-1, -2)}),
+        (180, {'k': 2, 'dims': (-1, -2)}),
+    ])
+    @pytest.mark.parametrize('scripted_affine', [True, False])
```

Comment:
Let's parametrize over `fn` directly: this will avoid the if/else, and also this will script the function only once instead of for each call.

We should change the other functions similarly


```suggestion
    @pytest.mark.parametrize('fn', (F.affine, scripted_affine))
```

(see my other comment for `scripted_affine`)

Reply:
done